File: app/src/main/java/com/treelzebub/umap/api/discogs/DiscogsApi.java
package com.treelzebub.umap.api.discogs;

import org.scribe.builder.api.DefaultApi10a;
import org.scribe.model.Token;

/**
 * Created by Tre Murillo 1/27/2015
 */
public class DiscogsApi extends DefaultApi10a {

    @Override
    public String getRequestTokenEndpoint() {
        return DiscogsConstants.REQUEST_TOKEN_URL;
    }

    @Override
    public String getAccessTokenEndpoint() {
        return DiscogsConstants.ACCESS_TOKEN_URL;
    }

    @Override
    public String getAuthorizationUrl(Token requestToken) {
        return DiscogsConstants.AUTHORIZATION_URL;
    }
}

File: app/src/main/kotlin/com/treelzebub/umap/api/discogs/DiscogsApi.kt
package com.treelzebub.umap.api.discogs

import com.treelzebub.umap.api.discogs.constants.*
import org.scribe.builder.api.DefaultApi10a
import org.scribe.model.Token

/**
 * Created by Tre Murillo on 5/28/15
 * Copyright(c) 2015 Level, Inc.
 */
public class DiscogsApi : DefaultApi10a() {

    override fun getRequestTokenEndpoint(): String {
        return REQUEST_TOKEN_URL
    }

    override fun getAccessTokenEndpoint(): String {
        return ACCESS_TOKEN_URL
    }

    override fun getAuthorizationUrl(requestToken: Token): String {
        return AUTHORIZATION_URL
    }
}
